file/file.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"os"
+)
+
+// Exists checks if specified file exists. It will read through symlinks.
```

Comment:
Sorry to nitpick.  Can we get a little chattier here about the semantics?  E.g. What about directories?  Document that a symlink which exists but points to a path that does not exist will return false here.  Pretend you don't know the implementation and are just reading the docs - what would you hope to see?

Reply:
Updated the docs to be a bit more explicit about behaviour. Does that seem thorough enough?